src/kubernetes.cc

```diff
@@ -1028,8 +1084,9 @@ void KubernetesReader::WatchNode(MetadataUpdater::UpdateCallback callback)
                           std::placeholders::_2, std::placeholders::_3));
   } catch (const json::Exception& e) {
```

Comment:
Do we need to handle the WatcherException here?

Reply:
No -- it will never be thrown in this thread, and exceptions don't propagate across threads.